src/python/openvino_genai/py_openvino_genai.pyi

```diff
@@ -0,0 +1,1703 @@
+        """
+    @typing.overload
+    def __init__(self, model: UNet2DConditionModel) -> None:
+        """
+        UNet2DConditionModel model
+                    UNet2DConditionModel class
+                    model (UNet2DConditionModel): UNet2DConditionModel model
+        """
+    def compile(self, device: str, **kwargs) -> None:
+        """
+                        Compiles the model.
+                        device (str): Device to run the model on (e.g., CPU, GPU).
+                        kwargs: Device properties.
+        """
+    def get_config(self) -> UNet2DConditionModel.Config:
+        ...
+    def infer(self, sample: openvino._pyopenvino.Tensor, timestep: openvino._pyopenvino.Tensor) -> openvino._pyopenvino.Tensor:
+        ...
+    def reshape(self, batch_size: int, height: int, width: int, tokenizer_model_max_length: int) -> UNet2DConditionModel:
+        ...
+    def set_adapters(self, adapters: AdapterConfig | None) -> None:
+        ...
+    def set_hidden_states(self, tensor_name: str, encoder_hidden_states: openvino._pyopenvino.Tensor) -> None:
+        ...
+class VLMPipeline:
+    """
+    This class is used for generation with VLMs
+    """
+    def __init__(self, models_path: os.PathLike, device: str, **kwargs) -> None:
+        """
+        device on which inference will be done
+                    VLMPipeline class constructor.
+                    models_path (str): Path to the folder with exported model files.
+                    device (str): Device to run the model on (e.g., CPU, GPU). Default is 'CPU'.
+                    kwargs: Device properties
+        """
+    def finish_chat(self) -> None:
+        ...
+    @typing.overload
+    def generate(self, prompt: str, images: list[openvino._pyopenvino.Tensor], generation_config: GenerationConfig, streamer: typing.Callable[[str], bool] | StreamerBase | None = None, **kwargs) -> DecodedResults:
+        """
+            Generates sequences for VLMs.
+        
+            :param prompt: input prompt
+            :type prompt: str
+        
+            :param images: list of images
+            :type images: List[ov.Tensor]
+        
+            :param generation_config: generation_config
+            :type generation_config: GenerationConfig or a Dict
+        
+            :param streamer: streamer either as a lambda with a boolean returning flag whether generation should be stopped
+            :type : Callable[[str], bool], ov.genai.StreamerBase
+        
+            :param kwargs: arbitrary keyword arguments with keys corresponding to GenerationConfig fields.
+            :type : Dict
+        
+            :return: return results in decoded form
+            :rtype: DecodedResults
+        """
+    @typing.overload
+    def generate(self, prompt: str, image: openvino._pyopenvino.Tensor, generation_config: GenerationConfig, streamer: typing.Callable[[str], bool] | StreamerBase | None = None, **kwargs) -> DecodedResults:
```

Comment:
maybe we need to name parameter with the same name for single / multiple images case? So, users can pass either list or a single image w/o different signatures

@Wovchena how do you think? what is the better name?

Reply:
1. I find overloads to be more readable. One function only casts args, the other function does the job. Having a common objects puts both cases to a single body.
2. Compile time C++ overloads insignificantly reduce overhead.

I prefer overloads

Reply:
The problem with overloads is that python has named args and people would need to distinguish image and images. That overweights my previous arguments. Let's stick to a single function with `image` arg